libc/stdio/stdio_indicator.c

```diff
@@ -0,0 +1,599 @@
+
+	errno = 0;
+	TEST_ASSERT_EQUAL(-1, ftell(pipe_stream));
+	TEST_ASSERT_EQUAL(ESPIPE, errno);
+
+	errno = 0;
+	TEST_ASSERT_EQUAL(-1, ftello(pipe_stream));
+	TEST_ASSERT_EQUAL(ESPIPE, errno);
+
+	fclose(pipe_stream);
+	close(pipefd[0]);
+	close(pipefd[1]);
+
+
+	const char *fifo_path = "myfifo";
+	mkfifo(fifo_path, 0666);
+
+	pid_t pid = fork();
+	TEST_ASSERT_NOT_EQUAL(-1, pid);
+
+	if (pid) {
+		int fifofd = open(fifo_path, O_WRONLY);
+		write(fifofd, "1234", 4);
+		close(fifofd);
+		exit(EXIT_SUCCESS);
+	}
+	else {
+		int status;
+		wait(&status);
+		TEST_ASSERT_EQUAL_INT(EXIT_SUCCESS, status);
+
+		int fifofd = open(fifo_path, O_RDWR);
+
+		TEST_ASSERT_NOT_EQUAL(-1, fifofd);
+
+		FILE *fifo_stream = fdopen(fifofd, "w+");
+
+		TEST_ASSERT_NOT_NULL(fifo_stream);
+
+		fflush(fifo_stream);
+
+		errno = 0;
+		TEST_ASSERT_EQUAL(-1, ftell(fifo_stream));
+		TEST_ASSERT_EQUAL(ESPIPE, errno);
+
+		errno = 0;
+		TEST_ASSERT_EQUAL(-1, ftello(fifo_stream));
+		TEST_ASSERT_EQUAL(ESPIPE, errno);
+
+
+		fclose(fifo_stream);
+		close(fifofd);
+	}
+
+
+	remove(fifo_path);
+}
+
+
+TEST_GROUP_RUNNER(stdio_ftell)
+{
+	RUN_TEST_CASE(stdio_ftell, correct_position_not_empty_file);
+	RUN_TEST_CASE(stdio_ftell, correct_position_empty_file);
+	RUN_TEST_CASE(stdio_ftell, bad_file_descriptor);
+	RUN_TEST_CASE(stdio_ftell, wrong_stream_type);
+}
+
+
+TEST_GROUP(stdio_getdelim);
+
+TEST_SETUP(stdio_getdelim)
+{
+}
+
+
+TEST_TEAR_DOWN(stdio_getdelim)
+{
+}
+
+
+TEST(stdio_getdelim, existing_delimiter_read_from_file_empty_or_simple_text)
+{
+	const char *filename = "Simple_text";
+	FILE *f = fopen(filename, "w+");
+
+	char *lineptr = NULL;
+	size_t len = 0;
+
+	TEST_ASSERT_EQUAL(-1, getdelim(&lineptr, &len, 'Q', f));
+	TEST_ASSERT_NOT_EQUAL(0, len);
+	TEST_ASSERT_NOT_NULL(lineptr);
+
+	fputs("IeIIeIII", f);
+	char *expected_first[3] = { "Ie", "IIe", "III" };
+
+
+	rewind(f);
+
+	/* Write to each cell of temp_result, compare it with expected_first */
+	int i = 0;
+	while (getdelim(&lineptr, &len, (int)'e', f) != -1) {
+		TEST_ASSERT_EQUAL_STRING(expected_first[i++], lineptr);
+	}
+
+	fclose(f);
+	free(lineptr);
+	remove(filename);
+}
+
+
+TEST(stdio_getdelim, existing_delimiter_read_from_long_text)
+{
+	const char *filename = "Long_text";
+	FILE *f = fopen(filename, "w+");
+
+	fputs(testdata_hugeStr, f);
+
+	char *lineptr = NULL;
+
+	fseek(f, 0, SEEK_SET);
+	size_t len = 0;
+
+	int i = 0;
+	while (getdelim(&lineptr, &len, (int)'e', f) != -1) {
+		i++;
+		TEST_ASSERT_NOT_NULL(lineptr);
+		TEST_ASSERT_GREATER_OR_EQUAL(1, len);
+		TEST_ASSERT_GREATER_OR_EQUAL(1, strlen(lineptr));
+	}
+
+	TEST_ASSERT_GREATER_THAN(10, i);
+
+	fclose(f);
+	free(lineptr);
+	remove(filename);
+}
+
+
+TEST(stdio_getdelim, too_small_buffer)
+{
+	char **buffer = NULL;
+	int n;
+	int *counter = &n;
+	*counter = 2;
+
+	FILE *f = fopen("decently_long_test", "w+");
+	fputs("RelativelyLongTextThatWillBeBiggerThanNInGetdelim", f);
+	rewind(f);
+
+	TEST_ASSERT_EQUAL(-1, getdelim(buffer, counter, '\n', f));
```

Comment:
Please resolve the following warnings, you can always make sure that no warnings were generated in `files changed` tab.
![Screenshot from 2023-11-20 18-13-09](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/77304431/0bcfe241-bafd-43c8-a592-14c3b88bd81b)